src/picongpu/include/particles/ionization/byCollision/ThomasFermi/AlgorithmThomasFermi.hpp

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright 2016-2017 Marco Garten
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "pmacc_types.hpp"
+#include "simulation_defines.hpp"
+#include "particles/traits/GetAtomicNumbers.hpp"
+#include "traits/attribute/GetChargeState.hpp"
+#include "algorithms/math/floatMath/floatingPoint.tpp"
+#include "particles/ionization/byCollision/ThomasFermi/TFFittingParameters.def"
+
+/** \file AlgorithmThomasFermi.hpp
+ *
+ * IONIZATION ALGORITHM for the Thomas-Fermi model
+ *
+ * - implements the calculation of average ionization degree and changes charge states
+ *   by decreasing the number of bound electrons
+ * - is called with the IONIZATION MODEL, specifically by setting the flag in @see speciesDefinition.param
+ */
+
+namespace picongpu
+{
+namespace particles
+{
+namespace ionization
+{
+
+    /** \struct AlgorithmThomasFermi
+     *
+     * \brief calculation for the Thomas-Fermi pressure ionization model
+     *
+     * This model uses local density and "temperature" values as input
+     * parameters. To be able to speak of a "temperature" an equilibrium state
```

Comment:
Phrasing suggestion: ~~To be able to speak of~~ -> A physical temperature requires a defined equilibrium state ...